pkg/cmd/roachtest/tests/admission_control_snapshot_overload.go

```diff
@@ -0,0 +1,198 @@
+)
+
+// This test sets up a 3-node CRDB cluster on 8vCPU machines, loads
+// it up with a large TPC-C dataset, and sets up a foreground load of kv95/1b.
+// The TPC-C dataset has a replication of factor of 1 and is used as the large
+// lump we bus around through snapshots -- snapshots that are send to the node
+// containing leaseholders serving kv95 traffic. Snapshots follow where the
+// leases travel, cycling through each node, evaluating performance isolation in
+// the presence of snapshots.
+//
+// TODO(irfansharif): This primarily stresses CPU; write an equivalent for IO.
+// TODO(irfansharif): This is a relatively long-running test (takes ~3.5hrs);
+// make it shorter.
+func registerSnapshotOverload(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:  "admission-control/snapshot-overload",
+		Owner: registry.OwnerAdmissionControl,
+		// TODO(irfansharif): After two weeks of nightly baking time, reduce
+		// this to a weekly cadence. This is a long-running test and serves only
+		// as a coarse-grained benchmark.
+		// 	Tags:    []string{`weekly`},
+		Cluster: r.MakeClusterSpec(4, spec.CPU(8)),
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			if c.Spec().NodeCount < 4 {
+				t.Fatalf("expected at least 4 nodes, found %d", c.Spec().NodeCount)
+			}
+
+			c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+			crdbNodes := c.Spec().NodeCount - 1
+			workloadNode := crdbNodes + 1
+			for i := 1; i <= crdbNodes; i++ {
+				startOpts := option.DefaultStartOpts()
+				startOpts.RoachprodOpts.ExtraArgs = append(startOpts.RoachprodOpts.ExtraArgs, fmt.Sprintf("--attrs=n%d", i))
+				c.Start(ctx, t.L(), startOpts, install.MakeClusterSettings(), c.Node(i))
+			}
+
+			db := c.Conn(ctx, t.L(), crdbNodes)
+			defer db.Close()
+
+			// Set a replication factor of 1 and pin replicas to n1 by default.
+			// For the active dataset (we use kv further below) we'll use a
+			// different config.
+			t.Status(fmt.Sprintf("configuring default zone and settings (<%s)", 30*time.Second))
+			{
+				if _, err := db.ExecContext(ctx,
+					"ALTER RANGE DEFAULT CONFIGURE ZONE USING num_replicas = 1, constraints = '[+n1]';",
+				); err != nil {
+					t.Fatalf("failed to configure zone for RANGE DEFAULT: %v", err)
+				}
+
+				// Defensive, since admission control is enabled by default. This
+				// test can fail if admission control is disabled.
+				setAdmissionControl(ctx, t, c, true)
+
+				// Set high snapshot rates.
+				if _, err := db.ExecContext(
+					ctx, "SET CLUSTER SETTING kv.snapshot_rebalance.max_rate = '256MiB'"); err != nil {
+					t.Fatalf("failed to set kv.snapshot_rebalance.max_rate: %v", err)
+				}
+				if _, err := db.ExecContext(
+					ctx, "SET CLUSTER SETTING kv.snapshot_recovery.max_rate = '256MiB'"); err != nil {
+					t.Fatalf("failed to set kv.snapshot_recovery.max_rate: %v", err)
+				}
+			}
+
+			t.Status(fmt.Sprintf("setting up prometheus/grafana (<%s)", 2*time.Minute))
```

Comment:
Nit: shouldn't the log match what the code is doing? (i.e., moving the `t.Status` call to `!t.SkipInit()` below).